13-Tenant Security Solution Setup/Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1

```diff
@@ -0,0 +1,639 @@
+    .PARAMETER SubscriptionId
+        Enter subscription id on which remediation need to perform.
+    .PARAMETER RemediationType
+        Select remediation type to perform from drop down menu.
+    .PARAMETER Path
+        Json file path which contain failed controls detail to remediate.
+    .PARAMETER PerformPreReqCheck
+        Perform pre requisities check to ensure all required module to perform roll back operation is available.
+    #>
+
+    param (
+        [string]
+        [Parameter(Mandatory = $true, HelpMessage="Enter subscription id for remediation")]
+        $SubscriptionId,
+
+        [Parameter(Mandatory = $true, HelpMessage = "Select remediation type")]
+        [ValidateSet("DisableAllowBlobPublicAccessOnStorage", "DisableAnonymousAccessOnContainers")]
+        [string]
+		$RemediationType,
+
+        [string]
+        [Parameter(Mandatory = $false, HelpMessage="Json file path which contain storage account detail to remediate")]
+        $Path,
+
+        [switch]
+        $PerformPreReqCheck
+    )
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "Starting to remediate anonymous access on containers of storage account from subscription [$($SubscriptionId)]..."
+    Write-Host $([Constants]::SingleDashLine)
+    
+    if($PerformPreReqCheck)
+    {
+        try 
+        {
+            Write-Host "Checking for pre-requisites..."
+            Pre_requisites
+            Write-Host $([Constants]::SingleDashLine)     
+        }
+        catch 
+        {
+            Write-Host "Error occured while checking pre-requisites. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)    
+            break
+        }
+    }
+
+    # Connect to AzAccount
+    $isContextSet = Get-AzContext
+    if ([string]::IsNullOrEmpty($isContextSet))
+    {       
+        Write-Host "Connecting to AzAccount..."
+        Connect-AzAccount
+        Write-Host "Connected to AzAccount" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # Setting context for current subscription.
+    $currentSub = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop -Force
+
+    Write-Host "Metadata Details: `n SubscriptionId: [$($SubscriptionId)] `n AccountName: [$($currentSub.Account.Id)] `n AccountType: [$($currentSub.Account.Type)]"
+    Write-Host $([Constants]::SingleDashLine)  
+    Write-Host "Starting with Subscription [$($SubscriptionId)]..."
+
+    Write-Host "`n"
+    Write-Host "*** To perform remediation for disabling anonymous access on containers user must have atleast contributor access on storage account of Subscription: [$($SubscriptionId)] ***" -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host "`n" 
+
+    Write-Host "Validating whether the current user [$($currentSub.Account.Id)] have the required permissions to run the script for Subscription [$($SubscriptionId)]..."
+    Write-Host "`n"
```

Comment:
At which step we are actually for permissions.

Reply:
Updated comments